doc/internals/estrin.qbk

```diff
@@ -0,0 +1,49 @@
+[section:estrin Estrin's method for polynomial evaluation]
+
+[h4 Synopsis]
+
+``
+#include <boost/math/tools/estrin.hpp>
+``
+
+   namespace boost { namespace math { namespace tools {
+ 
+   // Advanced interface: Use if you can preallocate a scratch buffer of size (coeffs.size() +1)/2:
+   // The scratch buffer size is *unchecked*, so use at your own risk!
```

Comment:
Since we know what size the buffer is supposed to be can we `assert(buffer.size() >= (N+1)/2)`?